src/test/java/com/contact/application/controller/ContactControllerTest.java

```diff
@@ -0,0 +1,136 @@
+    private UpdateController updateControllerMock = mock(UpdateController.class);
+    private DeleteController deleteControllerMock = mock(DeleteController.class);
+
+    private ContactController contactController;
+    private List<Contact> expectedContactList;
+    private Contact expectedContact;
+    private String expectedContactId;
+    private String invalidId;
+    @Before
+    public void setUp() {
+        testUtils = new TestUtils();
+        contactController = new ContactController(getControllerMock, postControllerMock, updateControllerMock, deleteControllerMock);
+        expectedContactId = testUtils.getUUID();
+        expectedContact = testUtils.getRandomContact(expectedContactId);
+        expectedContactList = Collections.singletonList(expectedContact);
+        invalidId = "invalid";
+
+        when(getControllerMock.getAllContacts()).thenReturn(expectedContactList);
+        when(getControllerMock.getContactById(eq(expectedContactId))).thenReturn(Optional.of(expectedContact));
+        when(getControllerMock.getContactById(eq(invalidId))).thenReturn(Optional.empty());
+
+        when(postControllerMock.createContact(eq(expectedContact))).thenReturn(Optional.of(expectedContactId));
+        when(postControllerMock.createContact(eq(null))).thenReturn(Optional.empty());
+
+        when(updateControllerMock.updateContact(eq(expectedContact), eq(expectedContactId))).thenReturn(Optional.of(expectedContact));
+        when(updateControllerMock.updateContact(eq(expectedContact), eq(invalidId))).thenReturn(Optional.empty());
+
+        when(deleteControllerMock.deleteContact(eq(expectedContact), eq(expectedContactId))).thenReturn(Optional.of(expectedContact));
+        when(deleteControllerMock.deleteContact(eq(expectedContact), eq(invalidId))).thenReturn(Optional.empty());
+    }
+
+    @Test
+    public void getAllContactsTest_ReturnValidList_AndReturnCorrectCode() {
+        HttpEntity<List<Contact>> entity = contactController.getAllContacts();
+        ResponseEntity responseEntity = (ResponseEntity) entity;
+        assertNotNull(entity.getBody());
+        assertEquals(entity.getBody(), expectedContactList);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+    }
+
+    @Test
+    public void getContactById_givenValidId_ShouldReturnValid_Contact() {
+        HttpEntity<Contact> entity = contactController.getContactById(expectedContactId);
+        ResponseEntity responseEntity = (ResponseEntity) entity;
+        assertNotNull(entity.getBody());
+        assertEquals(entity.getBody(), expectedContact);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+    }
+
+    @Test
+    public void getContactById_givenIncorrectId_ShouldNotReturnValid_Contact() {
+        HttpEntity<Contact> entity = contactController.getContactById(invalidId);
+        ResponseEntity responseEntity = (ResponseEntity) entity;
+        assertNotNull(entity.getBody());
```

Comment:
I am seeing a lot of repetition

Reply:
ok, will make that change.

Reply:
`contactController.getContactById` is just called twice, and I think it's okay to leave it as it, once it gets to be three or more times I will refactor.